site/content/contribute/developer-setup.md

```diff
@@ -0,0 +1,118 @@
+---
+title: "Developer setup"
+heading: "Set up your development environment"
+description: "Find out how to set up your development environment for building, running, and testing the Mattermost."
+weight: 1
+aliases:
+  - /contribute/server/developer-setup
+  - /contribute/more-info/server/developer-setup/
+  - /contribute/more-info/webapp/developer-setup/
+---
+
+Set up your development environment for building, running, and testing the Mattermost.
+
+{{<note "Warning" "icon-alert-outline">}}
+During the move to the monorepo these instructions will be in flux. Follow https://github.com/mattermost/mattermost-server/issues/22420 for updates.
+{{</note>}}
+
+{{<note "Note:">}}
+If you're developing plugins, see the plugin [developer setup]({{< ref "/integrate/plugins/developer-setup" >}}) documentation.
+{{</note>}}
+
+# Setup the Mattermost Server
+
+1. Install [Docker](https://www.docker.com/)
+
+2. Install [Go](https://go.dev/)
+
+3. Increse the number of available file descriptors. Update your shell's initialization script (e.g. `.bashrc` or `.zshrc`) and add the following:
+
+    ```sh
+    ulimit -n 8096
+    ```
+
+5. Fork https://github.com/mattermost/mattermost-server
```

Comment:
looks like this should be step 4 (and the rest of these steps should be renumbered)